### PR TITLE
Fix date parsing for base_players model to handle various date formats

### PR DESCRIPTION
**Pull request by fifthfrankie**
This PR addresses an issue with date parsing in the base_players model where certain date strings, such as years only (e.g., "2006"), were causing runtime errors due to incompatible format specifiers. The changes ensure robust date handling by accommodating multiple date formats, specifically:

- **Handles Full Dates:** Dates in the format '%b %d, %Y' (e.g., "Jan 01, 2000").
- **Handles Year-Only Dates:** Strings with only the year (e.g., "2006") are parsed by appending '-01-01' to form a complete date ("2006-01-01"), which is then cast to a DATE type.

**Changes Made:**

- Updated the SQL query in models/base/transfermarkt_scraper/base_players.sql:

> - Used CAST and STRPTIME functions to handle different date formats.
> - Added conditional checks to ensure that unsupported date formats are handled gracefully, defaulting to NULL when appropriate.

- Modified the logic to ensure compatibility with DuckDB and avoid errors related to unsupported functions like to_date.

**Checklist:**

- [x] Forked the repository.
- [x]  Set up the local environment following the [setup instructions](https://github.com/dcaribou/transfermarkt-datasets/compare/master...fifthfrankie:transfermarkt-datasets:fix-date-parsing?expand=1#-setup).
- [x]  Populated the data/raw directory as per the [data storage guidelines](https://github.com/dcaribou/transfermarkt-datasets/compare/master...fifthfrankie:transfermarkt-datasets:fix-date-parsing?expand=1#-data-storage).
- [x]  Modified the dbt project files to include robust date parsing logic in the base_players model.
- [x]  Tested changes locally to ensure all modifications work as intended without breaking existing functionalities.

**Testing:**

- Verified that the modified base_players model successfully parses various date formats without errors.
- Ran the full data preparation pipeline locally to confirm that all models and dependencies operate correctly post-change.

_If any further modifications or testing are required, please feel free to provide feedback, and I will address them promptly._

### CODE REVIEW

File: dbt/models/base/transfermarkt_scraper/base_players.sql [hunk 1]
```diff
@@ -34,14 +34,26 @@ select
...
+    --     end,
+    --     '%b %d, %Y'
+    -- )::date as date_of_birth,
+
+    case
+        when json_extract_string(json_row, '$.date_of_birth') not in ('N/A', 'null', '')
+        then 
+            case
+                when length(json_extract_string(json_row, '$.date_of_birth')) = 4
+                then CAST(json_extract_string(json_row, '$.date_of_birth') || '-01-01' AS DATE)  -- Assume first day of the year if only year is given
```
Comment (dcaribou) [inline dbt/models/base/transfermarkt_scraper/base_players.sql:51]:
```suggestion
                then cast(json_extract_string(json_row, '$.date_of_birth') || '-01-01' as date)  -- Assume first day of the year if only year is given
```

File: dbt/models/base/transfermarkt_scraper/base_players.sql [hunk 2]
```diff
@@ -34,14 +34,26 @@ select
...
 
-    strptime(
-        case
-            when json_extract_string(json_row, '$.date_of_birth') not in ('N/A', 'null', '')
-            then json_extract_string(json_row, '$.date_of_birth')
-            else null
-        end,
-        '%b %d, %Y'
-    )::date as date_of_birth,
+    -- strptime(
```
Comment (dcaribou) [inline dbt/models/base/transfermarkt_scraper/base_players.sql:37]: Could we remove the commented out part now?

**Review by dcaribou — CHANGES_REQUESTED** [review]
Thanks @fifthfrankie for your contribution and the amazing PR description ❤️ 

I left a couple of minor comments.

**Comment by fifthfrankie** [comment]
Apologies for the messy commits; a random timeout error cause branch to become outdated when pushing. All changes reverted and re-committed. Updated as per comments.

**Comment by dcaribou** [comment]
> Apologies for the messy commits; a random timeout error cause branch to become outdated when pushing. All changes reverted and re-committed. Updated as per comments.

No problem 😄 

Could you merge the latest changes from master? I pushed a small fix in the data earlier.

**Pull request merged**